sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml

```diff
@@ -0,0 +1,31 @@
+provider:
+  metadata:
+    name: Microsoft Teams Webhook Url
+    category: Messaging System
+    message: Make sure this Microsoft Teams Webhook Url gets revoked, changed, and removed from the code.
+
+  rules:
+    - rspecKey: S6721
+      id: ms-teams-webhook-urls
+      metadata:
+        name: Microsoft Teams Webhook Urls
+      examples:
+        - text: |
+            var webhookUrl = "https://companyname.webhook.office.com/webhookb2/5bf015e2-ce92-42a8-8e90-92552d6ef161@0c6aa5a3-6531-4e28-9c0b-33ba1a8aa1ff/IncomingWebhook/f7fb2308e5f14431ace5b7cd0e670e42/4563618c-b03b-4e80-b093-28bb4ff11de8";
+            var client = new HttpClient();
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            var content = new StringContent(adaptiveCardJson, System.Text.Encoding.UTF8, "application/json");
+            var response = await client.PostAsync(webhookUrl, content);
+          containsSecret: true
+          match: https://companyname.webhook.office.com/webhookb2/5bf015e2-ce92-42a8-8e90-92552d6ef161@0c6aa5a3-6531-4e28-9c0b-33ba1a8aa1ff/IncomingWebhook/f7fb2308e5f14431ace5b7cd0e670e42/4563618c-b03b-4e80-b093-28bb4ff11de8
+        - text: |
+            const NETWORKS = {
+              '1': 'https://outlook.office.com/webhook/6ee4efaa-986e-4b8e-8d6b-cca47c84f4a2@87ba1f9a-44cd-43a6-b008-6fdb45a5204e/TravisCI/9ad1eb07cda7411db33e74c3d676774e/0ad415e3-d66a-425c-9914-e9e52f2b81e1',
+            }
+          containsSecret: false
+      detection:
+        matching:
+          # Structure:
+          # https://tenant-name.webhook.office.com/webhookb2/guid@guid/IncomingWebhook/connectorId/guid
+          # An MS GUID is in the following format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX where X is a hex digit
+          pattern: "((https://)?[a-z0-9]+\\.webhook\\.office\\.com/webhookb2/[a-z0-9\\-]+@[a-z0-9\\-]+/IncomingWebhook/[a-z0-9]+/[a-z0-9\\-]+)"
```

Comment:
Needs a post filter to filter out 
``` 
https://yourorg.webhook.office.com/webhookb2/123@456/IncomingWebhook/789/000
https://abcXXX.webhook.office.com/webhookb2/AAAAAA@BBBBBBBB/IncomingWebhook/XXXXXXXXXX/YYYYYY
https://YOURTENANTNAME.webhook.office.com/webhookb2/xxxxxxx@yyyyyyyyy/IncomingWebhook/c88bf2bcfa8e498e9d67bfe192cf9a58/zzzzzzzzz
https://mycompany.webhook.office.com/webhookb2/6zyp5abc-95oe-qr7w-dpkz-ejcr2hubspr4@s1zzdh7u-z2e5-g1j9-j2qn-gxadreclbzia/IncomingWebhook/9w30n4elvrl9f844lra09hqqykuwfzoe/1i476glp-bi8u-8qu3-o76j-2u9vw0l6cbox
https://alelmsa.webhook.office.com/webhookb2/xxxx-xxxx-xxx-xxx@xxxxxxxx-xxxx-xxxx-xxxxxxx/IncomingWebhook/xxxx/xx
https://test.webhook.office.com/webhookb2/11111111-4444-4444-8444-cccccccccccc@22222222-4444-4444-8444-cccccccccccc/IncomingWebhook/33333333012222222222333333333344/44444444-4444-4444-8444-cccccccccccc
```
I suggest taking a look at the big-ass link below. It contains examples where the dummy part is the company (example, contoso, acmecorp, etc) subdomain, or repetitions of the same string in generam

https://sourcegraph.com/search?q=%28%28https%3A%5C%2F%5C%2F%29%3F%5Ba-z0-9%5D%2B%5C.webhook%5C.office%5C.com%5C%2Fwebhookb2%5C%2F%5Ba-z0-9%5C%5C-%5D%2B%40%5Ba-z0-9%5C%5C-%5D%2B%5C%2FIncomingWebhook%5C%2F%5Ba-z0-9%5D%2B%5C%2F%5Ba-z0-9%5C%5C-%5D%2B%29&patternType=standard&sm=1&groupBy=repo